src/Services/BaseService.php

```diff
@@ -0,0 +1,194 @@
+
+class BaseService
+{
+    /**
+     * Passed in data should be vetted and fully qualified from calling service class
+     * Expect to see some search helpers here as well.
+     */
+    private $table;
+    private $fields;
+    private $autoIncrements;
+
+    /**
+     * Default constructor.
+     */
+    public function __construct($table)
+    {
+        $this->table = $table;
+        $this->fields = sqlListFields($table);
+        $this->autoIncrements = self::getAutoIncrements($table);
+    }
+
+    /**
+     * buildInsertColumns
+     * Build an insert set and bindings
+     *
+     * @param array $passed_in
+     * @return array
+     */
+    protected function buildInsertColumns($passed_in = array())
+    {
+        $keyset = '';
+        $bind = array();
+        $result = array();
+
+        foreach ($passed_in as $key => $value) {
+            // Ensure auto's not passed in.
+            if (in_array($key, array_column($this->autoIncrements, 'Field'))) {
+                continue;
+            }
+            if (!in_array($key, $this->fields)) {
+                continue;
+            }
+            if ($value == 'YYYY-MM-DD' || $value == 'MM/DD/YYYY') {
+                $value = "";
+            }
+            if ($value === null || $value === false) {
+                $value = "";
+            }
+            $keyset .= ($keyset) ? ", `$key` = ? " : "`$key` = ? ";
+            $bind[] = ($value === null || $value === false) ? '' : add_escape_custom($value);
+        }
+
+        $result['set'] = $keyset;
+        $result['bind'] = $bind;
+
+        return $result;
+    }
+
+    /**
+     * buildUpdateColumns
+     * Build an update set and bindings
+     *
+     * @param array $passed_in
+     * @return array
+     */
+    protected function buildUpdateColumns($passed_in = array())
+    {
+        $keyset = '';
+        $bind = array();
+        $result = array();
+
+        foreach ($passed_in as $key => $value) {
+            if (in_array($key, array_column($this->autoIncrements, 'Field'))) {
+                continue;
+            }
+            if (!in_array($key, $this->fields)) {
+                // placeholder. could be for where clauses
+                $bind[] = ($value == 'NULL') ? "" : add_escape_custom($value);
+                continue;
+            }
+            if ($value == 'YYYY-MM-DD' || $value == 'MM/DD/YYYY') {
+                $value = "";
+            }
+            if ($value === null || $value === false) {
+                // in case unwanted values passed in.
+                continue;
+            }
+            $keyset .= ($keyset) ? ", `$key` = ? " : "`$key` = ? ";
+            $bind[] = ($value == 'NULL') ? "" : add_escape_custom($value);
+        }
+
+        $result['set'] = $keyset;
+        $result['bind'] = $bind;
+
+        return $result;
+    }
+
+    /**
+     * Verify pid exist
+     * @param $pid
+     * @return mixed
+     */
+    public function verifyPid($pid)
```

Comment:
Should we make something common for other id's its hardcoded and can be used only for PIDs. Maybe we can take the `id` and `table` name and then check :)

Reply:
That's fine and i'm sure there will other verifies we'll need. We just need to be careful with inserts, especially patient_data where it is possible to have duplicate patients but not pid